docs/draft/rest_api.md

```diff
@@ -1,201 +1,293 @@
-# First Draft for the REST API
+# Alternative draft for the REST api
 All calls and requests return data as JSON.
-## General Calls
-### Get current status of the transformer
-```
-GET status/
-```
-**returns:**
-  
-HTTP Response Code: 200 (always)
-  
+### Current status
+##### GET /status
+Get the current status of the transformer.
+*returns:* HTTP Response Code: 200 (always)
+
 ```json
 {
-  "state": "idle",
-  "current_transformations": [1,2,3],
-  "available_storage": 1000,
-  "total_storage": 10000,
-  "capabilities": ["aws", "kubernetes", "openstack"]
+    "status": "idle",
+    "available_storage": 1000,
+    "total_storage": 10000,
 }
 ```
-- *state*: current status of the transformer, values can either be `idle`, `transforming` or `error`
-- *current_transformations:* ids of the active or queued transformations (Array of unsigned integers)
-- *available_storage*: represents the free hard disk space in megabytes (Unsigned integer)
-- *total_storage*: represents the total storage of the hard disk (Unsigned integer)
-- *capabilities*: represents a list of the plugins loaded, i.e. the platforms which are supported by this instance of the transformer
-
-## Transformation specific calls
+- *status*: current status of the transformer, values can either be `idle`, `transforming` or `error`
+- *available_storage*: free hard disk storage in MB (integer)
+- *total_storage*: total storage of the hard disk (integer, in MB)
 
-***Error Note***: Every Call that requires an ID will return a Error 404 if the ID cannot be found.
+### Manage target platforms
 
-### Retrieve the result of the transformation
+##### GET /platforms
+Returns all target platforms which are available for transforming the CSAR.
 
-```
-GET transformation/{id}
-```
-  
-*Parameters*:
-- **id**: The id of the transformation for which the result should be retrieved.
-  
-**on success returns:**
-  
-HTTP Response Code: 200
+*returns:*
 ```json
-{
-  "id": 1,
-  "result": "/results/transform_1.zip"
-}
+[ 
+    "aws": {
```

Comment:
Btw this alternative rest api draft is the old version, but refactored and enriched. So don't worry, I did not only take one look at the outdated version